components/org.wso2.carbon.identity.oauth2.grant.organizationswitch/src/main/java/org/wso2/carbon/identity/oauth2/grant/organizationswitch/OrganizationSwitchGrant.java

```diff
@@ -183,6 +191,12 @@ private void checkOrganizationIsAllowedToSwitch(String currentOrgId, String swit
                 throw new IdentityOAuth2ClientException("Organization switch is only allowed for the organizations " +
                         "in the same branch.");
             }
+            if (!OrganizationSwitchGrantConstants.CONSOLE_APP_NAME.equals(appName) &&
+                    !getOrgApplicationManager().isApplicationSharedWithGivenOrganization(appID, currentOrgId,
```

Comment:
Let's add a comment saying, swithing is allowed only for the b2b application shared orgs.

Reply:
Fixed